x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -527,25 +465,18 @@ public void onFailure(Exception e) {
             );
 
             // Try adding state doc mapping
-            ActionListener<Boolean> resultsPutMappingHandler = ActionListener.wrap(
+            ActionListener<Void> getJobHandler = ActionListener.wrap(
                     response -> {
-                        addDocMappingIfMissing(AnomalyDetectorsIndex.jobStateIndexWriteAlias(), ElasticsearchMappings::stateMapping,
-                                state, jobUpdateListener);
+                        ElasticsearchMappings.addDocMappingIfMissing(AnomalyDetectorsIndex.jobStateIndexWriteAlias(),
```

Comment:
Do we still need to call this from here? Isn't it enough to call it only in `AutodetectProcessManager.openJob`?

Reply:
That one is updating the state index mappings.  The one I moved updates the results index mappings.

Since the state index mapping is simply `enabled=false` at the top level, in 7.x we don't need to update it at all - we know any state index that exists in 7.x is either a single type index from 6.x or else a typeless 7.x index.  But in 6.x we might, if a user has just upgraded from 5.4 where we had a multi-type state index.  Since this PR is being backported I didn't want to make a 7.x-only change in it.  But I can do that in a followup.